### ISSUE TITLE
Adopt spring-boot 2.x new and improved relaxed naming

### ISSUE THREAD

**Issue by kdvolder** [spring-projects/sts4#331]
See: https://github.com/spring-projects/sts4/issues/327

Summary. Since spring-boot 2.0 relaxed name rules have become much more logical and well-defined which opens the door to us finally properly supporting this.

The rules are very well explained here:
https://github.com/spring-projects/spring-boot/wiki/Relaxed-Binding-2.0

Adopting the rules will mean re-thinking how we index and lookup property names. 

I imagine we wouldn't index things based on the keys in the metadata (which contain hyphens) but rather based on the 'canonical' value of the key (lower case, all special chars removed).

Lookups should also use the canonical key. This means revisiting every place in the code that calls a lookup operation

- ensure that canonical key is passed rather than the data found in the yaml file
- make sure that caller is capable of dealing with discrepancies between the returned metadata items and their own (non-canonicalized version) of the key.

**Comment by kdvolder** [comment]
I finally got round again to taking a look a this. I've started reading and trying to really understand some of the materials around the new binding mechanics. On the wiki:

- https://github.com/spring-projects/spring-boot/wiki/Relaxed-Binding-2.0
- https://github.com/spring-projects/spring-boot/wiki/IDE-binding-feat
- https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-Configuration-Binding

This has raised some questions for me. I've put these questions here: 

https://github.com/spring-projects/spring-boot/issues/17570

As that seemed to be an issue created specifically around providing assistance to implementing precise rules around the name binding.

I'm putting this issue back on hold for the time being as I don't think its a good idea to dive into this major refactoring until I got some answers.
